homeassistant/components/bluetooth/match.py

```diff
@@ -74,7 +74,10 @@ def seen_all_fields(
     previous_match: IntegrationMatchHistory, advertisement_data: AdvertisementData
 ) -> bool:
     """Return if we have seen all fields."""
-    if not previous_match.manufacturer_data and advertisement_data.manufacturer_data:
+    if (
+        previous_match.manufacturer_data is None
+        and advertisement_data.manufacturer_data is not None
```

Comment:
`advertisement_data.manufacturer_data` is typed to `dict[int, bytes]` it can never be None unless there is a bug upstream